apps/meteor/app/reactions/server/setReaction.ts

```diff
@@ -51,40 +38,62 @@ async function setReaction(room: IRoom, user: IUser, message: IMessage, reaction
 	// 	return;
 	// }
 
-	const userAlreadyReacted =
-		message.reactions &&
-		Boolean(message.reactions[reaction]) &&
-		message.reactions[reaction].usernames.indexOf(user.username as string) !== -1;
-	// When shouldReact was not informed, toggle the reaction.
+	const reactedUsernamesForEmoji = message.reactions?.[reaction]?.usernames ?? [];
+
+	const userIndexInReactionList = reactedUsernamesForEmoji.indexOf(user.username as string);
+
+	const userAlreadyReacted = userIndexInReactionList !== -1;
+
 	if (shouldReact === undefined) {
 		shouldReact = !userAlreadyReacted;
 	}
 
-	if (userAlreadyReacted === shouldReact) {
+	if (shouldReact === userAlreadyReacted) {
 		return;
 	}
 
-	let isReacted;
+	let didReact;
 
 	if (userAlreadyReacted) {
```

Comment:
```suggestion
	if (userAlreadyReacted && message.reactions) {
```
This should be ok to validate + allows us to remove the lint + typecheck errors and see the tests running